### PR TITLE
Add documentation to Lib_modules and Vimpl

### PR DESCRIPTION
**Pull request by rgrinberg**
Also remove unused Lib_modules.have_artifacts function

Signed-off-by: Rudi Grinberg <rudi.grinberg@gmail.com>

### CODE REVIEW

File: src/lib_modules.mli [hunk 1]
```diff
@@ -1,21 +1,37 @@
...
+    true)] and when there is more than 1 module. *)
 val alias_module : t -> Module.t option
 
 val modules : t -> Module.Name_map.t
 
+(** Returns all the compatibility modules. These modules are compiled
+    separately *)
 val wrapped_compat : t -> Module.Name_map.t
 
+(** return the main module name if it exists. It exist for libraries with
```
Comment (emillon) [inline src/lib_modules.mli]:
```suggestion
(** Returns the main module name if it exists. It exist for libraries with
```

File: src/lib_modules.mli [hunk 2]
```diff
@@ -1,21 +1,37 @@
...
+   [(wrapped true)] or one module libraries. *)
 val main_module_name : t -> Module.Name.t option
 
+(** return only the virtual modules in the library *)
```
Comment (emillon) [inline src/lib_modules.mli]:
```suggestion
(** Returns only the virtual modules in the library *)
```

File: src/lib_modules.mli [hunk 3]
```diff
@@ -1,21 +1,37 @@
...
 val virtual_modules : t -> Module.Name_map.t
 
+(** All modules that need to be considered for installation. *)
 val installable_modules : t -> Module.t list
 
+(** For wrapped libraries, this is the user written entry module for the
+    library. For single module library's, it's the sole module in the library *)
```
Comment (emillon) [inline src/lib_modules.mli]:
```suggestion
    library. For single module libraries, it's the sole module in the library *)
```

**Comment by rgrinberg** [comment]
Merging as this is harmless

**Pull request merged**